lib/collection/cookie.js

```diff
@@ -157,15 +157,15 @@ _.assign(Cookie.prototype, /** @lends Cookie.prototype */ {
              * host name that is sending the Set-Cookie header. The SDK does not perform this check, but the underlying
              * client that actually sends the request could do it automatically.
              */
-            domain: options.domain,
+            domain: _.isString(options.domain) ? options.domain : undefined,
```

Comment:
On second thought, let's not drop this user data here. So, let's keep `domain`, `path`, and `extensions` as it is.